src/parser/datetime.rs

```diff
@@ -68,6 +70,92 @@ pub(crate) fn tokenize_interval(value: &str) -> Result<Vec<IntervalToken>, Parse
     Ok(toks)
 }
 
+fn tokenize_timezone(value: &str) -> Result<Vec<IntervalToken>, ParserError> {
+    let mut toks: Vec<IntervalToken> = vec![];
+    let mut num_buf = String::with_capacity(4);
+    /* If the timezone string has a colon, we need to parse all numbers naively, otherwise we need to parse long sequences of digits as [..hhhhmm] */
+    let split_nums: bool = value.find(':').unwrap_or(0) == 0;
+
+    fn parse_num(
+        toks: &mut Vec<IntervalToken>,
+        n: &str,
+        split_nums: bool,
+        idx: usize,
+    ) -> Result<(), ParserError> {
+        if n.is_empty() {
+            return Ok(());
+        }
+
+        if n.len() <= 2 || split_nums == false {
+            let token = IntervalToken::Num(n.parse().map_err(|e| {
+                ParserError::ParserError(format!(
+                    "Unable to parse value as a number at index {}: {}",
+                    idx, e
+                ))
+            })?);
+            toks.push(token);
+        } else {
+            /* Parse the string as two numbers */
+            let first = &n[..(n.len() - 2)];
+            let second = &n[(n.len() - 2)..];
+
+            toks.push(IntervalToken::Num(first.parse().map_err(|e| {
+                ParserError::ParserError(format!(
+                    "Unable to parse value {} as a number at index {}: {}",
+                    first, idx, e
+                ))
+            })?));
+            toks.push(IntervalToken::Num(second.parse().map_err(|e| {
+                ParserError::ParserError(format!(
+                    "Unable to parse value {} as a number at index {}: {}",
+                    second, idx, e
+                ))
+            })?));
+        }
+
+        Ok(())
+    };
+    for (i, chr) in value.chars().enumerate() {
+        match chr {
+            '-' => {
+                parse_num(&mut toks, &num_buf, split_nums, i)?;
+                num_buf.clear();
+                toks.push(IntervalToken::Dash);
+            }
+            ' ' => {
+                parse_num(&mut toks, &num_buf, split_nums, i)?;
+                num_buf.clear();
+                toks.push(IntervalToken::Space);
+            }
+            ':' => {
+                parse_num(&mut toks, &num_buf, split_nums, i)?;
+                num_buf.clear();
+                toks.push(IntervalToken::Colon);
+            }
+            '+' => {
+                parse_num(&mut toks, &num_buf, split_nums, i)?;
+                num_buf.clear();
+                toks.push(IntervalToken::Plus);
+            }
+            chr if chr.is_digit(10) => num_buf.push(chr),
+            chr if chr.is_ascii_alphabetic() => {
+                parse_num(&mut toks, &num_buf, split_nums, i)?;
+                let substring = &value[i..];
```

Comment:
I'm confused about what this `parse_num()` is doing inside of an alphabetic part?

Reply:
you have to clear the number buffer before you do other tokenizing

e.g. in the string "5z"